src/Compilers/CSharp/Test/Symbol/Symbols/TypeTests.cs

```diff
@@ -30,13 +31,29 @@ public void Bug18280()
 
             string code = "class C {  int " + brackets + @" x; }";
 
-            var compilation = CreateCompilation(code);
+            var compilation = CreateCompilation(code, options: WithNonNullTypesTrue());
             var c = compilation.GlobalNamespace.GetTypeMembers("C")[0];
             var x = c.GetMembers("x").Single() as FieldSymbol;
-            var arr = x.Type;
+            var type = x.Type;
+            type.GetHashCode();
+            foreachArray(x => Assert.Equal(NullableAnnotation.NotAnnotated, x.ElementTypeWithAnnotations.NullableAnnotation));
+            type = type.SetObliviousNullabilityForReferenceTypes();
+            foreachArray(x => Assert.Equal(NullableAnnotation.Oblivious, x.ElementTypeWithAnnotations.NullableAnnotation));
 
-            arr.GetHashCode();
-            arr.SetObliviousNullabilityForReferenceTypes();
+            void foreachArray(Action<ArrayTypeSymbol> action)
+            {
+                var arrayType = (ArrayTypeSymbol)type;
+                do
+                {
+                    var next = arrayType.ElementType as ArrayTypeSymbol;
+                    if (next is object)
```

Comment:
Is this check needed?

Reply:
Yes. The final element type is  `int` and this method is iterating over arrays.

Reply:
Ah I missed that. In that case, consider having the `Action` check `x.NullableAnnotation` rather than `x.Element.NullableAnnotation`.